vdr/doc/resolvers.go

```diff
@@ -190,6 +190,19 @@ func (r KeyResolver) ResolveAssertionKeyID(id did.DID) (ssi.URI, error) {
 	return ExtractAssertionKeyID(*doc)
 }
 
+// ResolveKeyAgreementKey resolves the public key of the first valid KeyAgreement of an indicated DID document in the current state.
+// If the document has no KeyAgreements, types.ErrKeyNotFound is returned.
+func (r KeyResolver) ResolveKeyAgreementKey(id did.DID) (crypto.PublicKey, error) {
+	doc, _, err := r.Store.Resolve(id, nil)
+	if err != nil {
+		return ssi.URI{}, err
+	}
+	if len(doc.KeyAgreement) == 0 {
+		return nil, types.ErrKeyNotFound
+	}
+	return doc.KeyAgreement[0].PublicKey()
```

Comment:
I think I've mentioned this before. In theory a DID could contain multiple public keys that can be used for key agreements. I'm fine with picking the first one but afaic it should be in the RFC

Reply:
Will do